File: fastapi_migration/frontend/src/types/report.ts
/**
 * Types related to genetic reports
 */

// Report generation request
export interface ReportRequest {
  file_hash?: string;
  analysis_id?: string;
  report_type: string;
  include_raw_data?: boolean;
}

// Report generation response
export interface ReportResponse {
  report_id: string;
  status: string;
  message: string;
  processing_time?: number;
  download_url: string;
  cached: boolean;
}

// Report metadata
export interface ReportMetadata {
  report_id: string;
  created_at: string;
  report_type: string;
  file_hash?: string;
  analysis_id?: string;
  download_url: string;
}

// Report summary for listings
export interface ReportSummary {
  report_id: string;
  created_at: string;
  report_type: string;
  download_url: string;
}

// User reports listing
export interface UserReportsList {
  reports: ReportSummary[];
}

// Report type options
export enum ReportType {
  STANDARD = "standard",
  MARKDOWN = "markdown"
}

// Report state for the application
export interface ReportState {
  generating: boolean;
  reportId?: string;
  reportType: ReportType;
  downloadUrl?: string;
  error?: string;
  recentReports: ReportSummary[];
}

File: fastapi_migration/backend/app/schemas/report.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime

# Report request schemas
class ReportRequest(BaseModel):
    file_hash: Optional[str] = None
    analysis_id: Optional[str] = None
    report_type: str = "markdown"  # markdown or standard
    include_raw_data: bool = False

# Report response schemas
class ReportResponse(BaseModel):
    report_id: str
    status: str
    message: str
    processing_time: Optional[float] = None
    download_url: str
    cached: bool = False

class ReportMetadata(BaseModel):
    report_id: str
    created_at: datetime
    report_type: str
    file_hash: Optional[str] = None
    analysis_id: Optional[str] = None
    download_url: str
    
    class Config:
        from_attributes = True

# Report listing schemas
class ReportSummary(BaseModel):
    report_id: str
    created_at: datetime
    report_type: str
    download_url: str

class UserReportsList(BaseModel):
    reports: List[ReportSummary] = []